### ISSUE TITLE
parse for @username in markdown fields on OSM, and link to their user page

### ISSUE THREAD

**Issue by mikelmaron** [openstreetmap/openstreetmap-website#1111]
In changeset discussions, diary entries, or elsewhere, support github style @ link within markdown

**Comment by emacsen** [comment]
I think you mean display names (OSM usernames are technically numbers). But Display names can have spaces in them, which could lead to confusion, so you really need a way to override this behavior.

**Comment by pnorman** [comment]
I don't think this is possible, for the space reason. As an example, consider `foo @bar baz foo` where there exist users `bar` and `bar baz`. What do you link to? How do you explain this to users in a non-confusing way?

**Comment by danstowell** [comment]
FWIW the Taking Manager already has an approach to this. They use square brackets for names with spaces in, for example @[Jo Schmo]. So yes it's possible, though obviously the space thing makes it a bit clunkier.
https://github.com/hotosm/osm-tasking-manager2/issues/369

**Comment by mikelmaron** [comment]
@danstowell that sounds like it would work.

**Comment by emacsen** [comment]
To do this, you may want to look at the work on the reporting branch, where I directed our GSoC student to do forward and reverse linking of OSM elements (node, way, relation) and changesets to markdown syntax.

**Comment by tomhughes** [comment]
@emacsen what is happening with that branch? am I ever getting a PR for it?

**Comment by emacsen** [comment]
@tomhughes that should have happened during the summer. I thought you were evaluating it. I'm fairly sure we discussed on IRC.

**Comment by tomhughes** [comment]
@emacsen I was asked for my comments on the branch yes, but no PR was ever opened as far as I can recall.

**Comment by emacsen** [comment]
@tomhughes this may be more appropriate for a separate issue, but yes I agree I'd love to see this code pulled into the main branch

**Comment by andrewharvey** [comment]
+ send a notification to the user who was mentioned.

I'd like to do this when commenting on a suspicious changeset by user A, who changed something user B originally contributed. By doing @ B I can loop them into to make them aware of the change and ask for any further details.

**Comment by mikelmaron** [comment]
@andrewharvery yes I had that in mind when opening this ticket --- 5 years ago! changeset and diary comments are already set up to send notifications to the editor, and anyone subscribed to the comments, so that's not hard. main thing is parsing for the usernames (also not that hard). of course if it was so easy, it would be done...

**Comment by danieldegroot2** [comment]
Related #693 / #844

**Comment by 1ec5** [comment]
#6518 implemented the @-username syntax in Markdown fields (diary entries, diary comments, user profile descriptions), and #6597 took care of changeset comments, which are plain text.

> Plus send a notification to the user who was mentioned.

#6636

**Comment by pablobm** [comment]
> https://github.com/openstreetmap/openstreetmap-website/pull/6518 implemented the @-username syntax in Markdown fields (diary entries, diary comments, user profile descriptions), and https://github.com/openstreetmap/openstreetmap-website/pull/6597 took care of changeset comments, which are plain text.

Actually, I just noticed this is not correct. Autolinking (for @-mentions as well as links to map elements, etc) was implemented only for the types of comment that we internally tag as "text": notes, changesets, and their respective comments. It doesn't apply to those we tag as "markdown": diary entries, diary comments, direct messages, etc.

I have left a question at https://github.com/openstreetmap/openstreetmap-website/pull/6518#issuecomment-4593594936, asking for the rationale.

**Comment by pablobm** [comment]
Created a new issue with more detail at https://github.com/openstreetmap/openstreetmap-website/issues/7122

**Comment by hlfan** [comment]
> Actually, I just noticed this is not correct.

Well at the time it was correct, but inconsistencies between linkify and auto_link lead to the revert in #6674.
